akka-sample-sharding-scala/src/main/scala/sample/sharding/Device.scala

```diff
@@ -1,54 +1,67 @@
 package sample.sharding
 
-import akka.actor._
+import akka.actor.typed.{ ActorRef, ActorSystem, Behavior }
+import akka.actor.typed.scaladsl.Behaviors
+import akka.cluster.sharding.typed.scaladsl.{ ClusterSharding, Entity, EntityTypeKey }
 
-/**
-  * This is just an example: cluster sharding would be overkill for just keeping a small amount of data,
-  * but becomes useful when you have a collection of 'heavy' actors (in terms of processing or state)
-  * so you need to distribute them across several nodes.
-  */
-object Device {
-  sealed trait Command extends Message
-
-  case class RecordTemperature(deviceId: Int, temperature: Double)
-      extends Command
-
-  case class GetTemperature(deviceId: Int) extends Command
-
-  case class Temperature(deviceId: Int,
-                         average: Double,
-                         latest: Double,
-                         readings: Int)
-      extends Message
-
-  def props(): Props =
-    Props(new Device)
-}
-class Device extends Actor with ActorLogging {
-  import Device._
-
-  override def receive = counting(Vector.empty)
-
-  def counting(values: Vector[Double]): Receive = {
-    case RecordTemperature(id, temp) =>
-      val temperatures = values :+ temp
-      log.info(
-        s"Recording temperature $temp for device $id, average is ${average(temperatures)} after " +
-          s"${temperatures.size} readings"
-      )
-      context.become(counting(temperatures))
-
-    case GetTemperature(id) =>
-      val reply =
-        if (values.isEmpty)
-          Temperature(id, Double.NaN, Double.NaN, 0)
-        else
-          Temperature(id, average(values), values.last, values.size)
-      sender() ! reply
+object Aggregator {
 
-  }
+  sealed trait Command
+  final case class Aggregate(stationId: Int, deviceId: Int, data: List[Double]) extends Command
+  final case class GetAverage(stationId: Int, deviceId: Int, replyTo: ActorRef[Guardian.AggregateData]) extends Command
+  final case class GetHiLow(stationId: Int, deviceId: Int, replyTo: ActorRef[Guardian.AggregateData]) extends Command
```

Comment:
AggregateData should maybe be defined here in the Aggregator, as a response message, but that would require message adapter or context.ask in the Guardian. We can keep it simple and use shared messages for those replies.